addons/report/models/report.py

```diff
@@ -142,11 +142,34 @@ def get_html(self, docids, report_name, data=None):
                 'docs': docs,
             }
             return self.render(report.report_name, docargs)
+            
     @api.model
     def get_pdf(self, docids, report_name, html=None, data=None):
-        """This method generates and returns pdf version of a report.
         """
-
+        This method generates and returns pdf version of a report. It also generate 
+        the corresponding attachment.
+        """
+        attachment_callback_data = {}
+        pdf_report = self._generate_pdf_report(docids,report_name, html=html, data=data, report_attachment_callback_data=attachment_callback_data)
+        
+        if attachment_callback_data: #If values were inserted into the dictionary
```

Comment:
on line 152 you assign an empty dict on this variable and it is never set to something else, so the whole if statement will always be ignored